lms/djangoapps/teams/views.py

```diff
@@ -368,6 +368,9 @@ def post(self, request):
             }, status=status.HTTP_400_BAD_REQUEST)
         else:
             team = serializer.save()
+            if not has_access(request.user, 'staff', course_key):
```

Comment:
The only method I found to return the discussion admin/moderator/ta status is lms/djangoapps/django_comment_client/utils.py, get_role_ids. It returns the IDs of the users with the privileged roles (and this is what the JavaScript method DiscussionUtil.isPrivilegedUser consumes). It is OK for teams to import from another LMS django app. You and @peter-fogg should collaborate to create a helper method that makes this boolean "isPrivileged" check a little nicer (Peter needs it for #9169).